Palindrome check

shrutivanw opened the pull request:
CS Fundamentals' assignment to implement a function that checks if the input string is a palindrome and accordingly returns true or false.

shrutivanw committed:
Palindrome check

CS Fundamentals' assignment to implement a function that checks if the input string is a palindrome and accordingly returns true or false.

Hamled commented on README.md:
```diff
@@ -1 +1,10 @@
+**Note**: Palindrome is a word, phrase or sentence that reads the same backwards as it does forwards. e.g. "madam"
+* Share and explain the time and space complexities for your solution.
+* At minimum, your implementation should pass the basic tests.
+**Note**: Do not use Ruby provided functionality for `.reverse` and `.reverse!`. You may use `.length`.
```
We should have an extra blank line above this so that the note goes into its own paragraph.

Hamled commented on specs/palindrome_check_spec.rb:
```diff
@@ -0,0 +1,84 @@
+
+describe "palindrome" do
+  describe "basic tests" do
+    describe "test 1" do
```
Same as the other repos, we should probably remove the `describe` blocks that are immediate parents of individual tests.

Hamled commented on specs/palindrome_check_spec.rb:
```diff
@@ -0,0 +1,84 @@
+
+  # if it's a string parameter, check for empty
+  describe "empty string input test" do
+    it "empty string reads the same backwards or forwards" do
```
I think this and the nil test should be moved into the edge cases section.

Hamled commented on lib/palindrome_check.rb:
```diff
@@ -0,0 +1,6 @@
+# Return true if the string is a palindrome. Return false otherwise.
+def palindrome_check(my_phrase)
+  puts "NOT IMPLEMENTED"
+  return true
```
How would you feel about doing `raise NotImplementedError` here instead?

Best practice I think is generally to avoid `puts` statements and other things that will print to the console for code being tested, because that output kind of gets randomly interspersed with the test runner's output.

shrutivanw committed:
Incorporating good feedback

- Made the method template raise NotImplementedError
- Moved empty and nil test cases to be within edge cases
- Removed inner-most describe for individal tests
- Made it so that readme shows line spacing correctly.

shrutivanw commented:
Thank you, @Hamled PR updated to incorporate feedback.
